powerfulseal/k8s/k8s_inventory.py

```diff
@@ -47,6 +47,17 @@ def is_fresh(self, when):
             return False
         return True
 
+    def get_regex_namespaces(self, regex):
+        regex_no_star = regex.replace("*", '')
+        namespaces = []
+        for item in self.k8s_client.list_namespaces():
+            m = re.match(regex, item.metadata.name)
+            if m:
+                stop_letter, start_letter = m.span()
+                if abs(stop_letter - start_letter) == len(regex_no_star):
```

Comment:
I'm not 100% sure what this check achieves?

Reply:
My intent was to be able to see if the regex string I gave matches the item.metadata.name. I think there is a better way to do this but I'm not 100% sure. 

In the above code, re.match returns the first and last character it matched. I ran into scenarios in my testing that it would match part of the string but not the whole string and give a false "match". I needed to be sure that the length of the matched string is the same length as the regex string without the star character.

Reply:
Oh I see. `re.match` is supposed to match substrings. If you want to match the exact string, you can specify the beginning and end of lines, for example `^something$`.

Reply:
I was trying to follow the similar way that the filters section works 
Example)
```
filters:
      - property:
          name: "name"
          value: "openshift-*"
```

Definitely open to not using re.match, if you can point me to how you did it for filters or know a better way to match this type of regex. I know this isn't the cleanest way

Reply:
I refactored a little to use re.compile and re.search instead. it looks a little cleaner, thoughts?